umake/ui/cli/__init__.py

```diff
@@ -157,6 +157,60 @@ def mangle_args_for_default_framework(args):
     return result_args
 
 
+def list_frameworks(listing, detailled=True):
+    if listing == "all" or listing == "available":
+        # Sort the categories to prevent a random list at each new program execution
+        for category in sorted(BaseCategory.categories.values(), key=lambda cat: cat.prog_name):
+            # The category must have at least one framework
+            if len(category.frameworks) == 0:
+                continue
+
+            if detailled:
+                print("{}: {}".format(category.prog_name, category.description))
+            else:
+                print("{}:".format(category.prog_name))
+
+            # Sort the frameworks to prevent a random list at each new program execution
+            for framework in sorted(category.frameworks.values(), key=lambda fram: fram.prog_name):
+                # Framework doesn't be removal
+                if framework.only_for_removal:
+                    continue
+
+                if listing == "available":
+                    if not framework.is_installable:
+                        continue
+
+                if detailled:
+                    # Attribute 'description' of 'framework' contains if the category is default
+                    print("  {}: {} ".format(framework.prog_name, framework.description), end="")
+                else:
+                    if framework.is_category_default:
+                        print("  {} (default) ".format(framework.prog_name, framework.description), end="")
+                    else:
+                        print("  {} ".format(framework.prog_name, framework.description), end="")
+
+                if not framework.is_installable:
+                    print("(not installable on this machine)", end="")
+                elif framework.is_installed:
+                    print("(installed)", end="")
+
+                print()
+    elif listing == "installed":
+        # Sort the categories to prevent a random list at each new program execution
+        for category in sorted(BaseCategory.categories.values(), key=lambda cat: cat.prog_name):
+            # Sort the frameworks to prevent a random list at each new program execution
+            for framework in sorted(category.frameworks.values(), key=lambda fram: fram.prog_name):
+                if framework.is_installed:
+                    if detailled:
+                        print("{}: {}".format(framework.prog_name, framework.description))
+                        print("  (path: {})".format(framework.install_path))
+                    else:
+                        print("{} (path: {})".format(framework.prog_name, framework.install_path))
+    else:
+        print("Wrong command")
+        exit(0)
```

Comment:
You are right, this was mainly a prototype function to test it quickly. In the future, I will think about what you said :)